src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,24 +1,109 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 /**
  * <p>Реалізувати свою HashMap, а саме методи `put(K key, V value)`, `getValue()` та `getSize()`.
  * Дотриматися основних вимог щодо реалізації мапи (initial capacity, load factor, resize...)
  * За бажанням можна реалізувати інші методи інтрефейсу Map.</p>
  */
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+    private int size;
+    private int threshold;
+    private Node<K, V>[] elements;
+
+    public MyHashMap() {
+        elements = new Node[DEFAULT_CAPACITY];
+        threshold = (int) (DEFAULT_CAPACITY * LOAD_FACTOR);
+        size = 0;
+    }
 
     @Override
     public void put(K key, V value) {
-
+        int index = getIndex(getHashCode(key));
+        if (elements[index] == null) {
+            elements[index] = new Node<>(key, value, null);
+        } else {
+            Node<K, V> temp = elements[index];
+            while (temp != null
+                    && (temp.next != null || Objects.equals(key, temp.key))) {
+                if (Objects.equals(key, temp.key)) {
+                    temp.value = value;
+                    return;
+                }
+                temp = temp.next;
+            }
+            temp.next = new Node<>(key, value, null);
+        }
+        size++;
+        checkSize();
     }
 
     @Override
     public V getValue(K key) {
+        int index = key == null ? 0 : getIndex(getHashCode(key));
+        Node<K, V> temp = elements[index];
+        while (temp != null) {
+            if (Objects.equals(temp.key, key)) {
+                return temp.value;
+            }
+            temp = temp.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int getHashCode(K key) {
+        if (key == null) {
+            return 0;
+        } else {
+            int hashCode = key.hashCode();
+            return hashCode ^ (hashCode >>> DEFAULT_CAPACITY);
+        }
+    }
+
+    private int getIndex(int hash) {
```

Comment:
```suggestion
    private int getIndex(K key) {
```
it would be more logical if you take key as input and return its position

Reply:
Yes, totally agree. I've changed it